src/utils/cairoDataTypes/felt.ts

```diff
@@ -9,7 +17,7 @@ import { encodeShortString, isShortString, isText } from '../shortString';
  */
 export function CairoFelt(it: BigNumberish): string {
   // BN or number
-  if (isBigInt(it) || (typeof it === 'number' && Number.isInteger(it))) {
+  if (isBigInt(it) || (isNumber(it) && Number.isInteger(it))) {
     return it.toString();
   }
   // string text
```

Comment:
The string handling look like it could be  restructured similar to `setidentifier()` from `src/utils/provider.ts`.

```
if (isString()) {
  // hex string
  if (isHex()) return ...
  
  // string number ...
  if (isStringWholeNumber()) return ...
  
  -- this now covers the conditionals from isText() so id doesn't need to be used
  //string text
  ...
}
```